docs: Improve magento2 quickstart and remember composer auth (#5932)
ddev/ddev: docs/content/users/quickstart.md

=== "Magento 2"

>     Normal details of a Composer build for Magento 2 are on the [Magento 2 site](https://experienceleague.adobe.com/docs/commerce-operations/installation-guide/composer.html). You must have a public and private key to install from Magento’s repository. When prompted for “username” and “password” in `composer create`, it’s asking for your public key as "username" and private key as "password".

Reviewer: Perhaps we can also add a tip here about credentials from https://ddev.readthedocs.io/en/stable/users/extend/in-container-configuration/ ?

```
mkdir -p ~/.ddev/homeadditions/.composer && ln -s ~/.composer/auth.json ~/.ddev/homeadditions/.composer/auth.json
```

I have set this up once and for all and I don't need to enter these credentials for new `magento2` projects.

Author: Reasonable to add. This is done automatically by this quickstart (putting in project homeadditions). 

Also, they provide an auth.json.example which could be used here.

Author: Added.

    ```bash
    SITENAME=ddev-magento2
    mkdir -p ${SITENAME} && cd ${SITENAME}
    ddev config --project-type=magento2 --php-version=8.2 --database=mariadb:10.6 --docroot=pub --disable-settings-management --upload-dirs=media
    ddev config --web-environment-add=COMPOSER_HOME="/var/www/html/.ddev/homeadditions/.composer"
    ddev get ddev/ddev-elasticsearch
    ddev start

>     ddev composer create --repository=https://repo.magento.com/ magento/project-community-edition --no-dev -y

Reviewer: `--no-dev` is not needed here because people need dev dependencies to run things like phpunit tests.

```suggestion
    ddev composer create --repository=https://repo.magento.com/ magento/project-community-edition -y
```

Author: Unfortunately it creates such a huge vendor directory that it works against itself. I'm pretty sure people can sort out what to do.  I added that because we couldn't even upload the resultant code to GitHub release. However, your point is still valid.

    rm -f app/etc/env.php
    echo "/auth.json" >.ddev/homeadditions/.composer/.gitignore

    # Change the base-url below to your project's URL
    ddev magento setup:install --base-url="https://${SITENAME}.ddev.site/" --cleanup-database --db-host=db --db-name=db --db-user=db --db-password=db --elasticsearch-host=elasticsearch --search-engine=elasticsearch7 --elasticsearch-port=9200 --admin-firstname=Magento --admin-lastname=User --admin-email=user@example.com --admin-user=admin --admin-password=Password123 --language=en_US

    ddev magento deploy:mode:set developer
    ddev magento module:disable Magento_TwoFactorAuth Magento_AdminAdobeImsTwoFactorAuth
    ddev config --disable-settings-management=false
    ddev launch
    ```

    Change the admin name and related information as needed.

>     The admin login URL is specified by `frontName` in app/etc/env.php.

Reviewer: ```suggestion
    The admin login URL is specified by `frontName` in `app/etc/env.php`.
```

>     You may want to add the [Magento 2 Sample Data](https://experienceleague.adobe.com/docs/commerce-operations/installation-guide/next-steps/sample-data/composer-packages.html) with

Reviewer: ```suggestion
    You may want to add the [Magento 2 Sample Data](https://experienceleague.adobe.com/docs/commerce-operations/installation-guide/next-steps/sample-data/composer-packages.html) with:
```

>     Normal details of a Composer build for Magento 2 are on the [Magento 2 site](https://devdocs.magento.com/guides/v2.4/install-gde/composer.html). You must have a public and private key to install from Magento’s repository. When prompted for “username” and “password” in `composer create`, it’s asking for your public and private keys.

Author: ```suggestion
    Normal details of a Composer build for Magento 2 are on the [Magento 2 site](https://experienceleague.adobe.com/docs/commerce-operations/installation-guide/composer.html). You must have a public and private key to install from Magento’s repository. When prompted for “username” and “password” in `composer create`, it’s asking for your public key as "username" and private key as "password".
```

    ```bash
    SITENAME=ddev-magento2
    mkdir -p ${SITENAME} && cd ${SITENAME}

>     ddev config --project-type=magento2 --php-version=8.1 --database=mariadb:10.6 --docroot=pub --disable-settings-management

Author: ```suggestion
    ddev config --project-type=magento2 --php-version=8.1 --database=mariadb:10.6 --docroot=pub --disable-settings-management --upload-dirs=media
```

    ddev config --web-environment-add=COMPOSER_HOME="/var/www/html/.ddev/homeadditions/.composer"
    ddev get ddev/ddev-elasticsearch
    ddev start
    ddev composer create --repository=https://repo.magento.com/ magento/project-community-edition -y
    rm -f app/etc/env.php

>     echo "/auth.json" >.ddev/homeadditions/.gitignore

Author: ```suggestion
    echo "/auth.json" >.ddev/homeadditions/.composer/.gitignore
```
